ui/app/src/modules/Content/Dependencies/DependencySelection.tsx

```diff
@@ -295,8 +295,8 @@ function SelectionList(props: SelectionListProps) {
 
   const paths = items ? items.map((item) => item.path) : uris;
 
-  const isAllChecked = useMemo(() => !paths?.some((path) => !checked[path]), [paths, checked]);
-  const isIndeterminate = useMemo(() => paths?.some((path) => checked[path] && !isAllChecked), [
+  const isAllChecked = useMemo(() => !paths?.some((path) => !checked?.[path]), [paths, checked]);
+  const isIndeterminate = useMemo(() => paths?.some((path) => checked?.[path] && !isAllChecked), [
```

Comment:
This is wasteful. Check for checked being null before iterating and return immediately without looping when nullish.